pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2839,32 +2883,56 @@ func maybeResizeDisk(diskfile string, maxsizebytes uint64) error {
 	return err
 }
 
-// getCloudInitData : returns decrypted cloud-init data
-func getCloudInitData(config types.DomainConfig) (*string, error) {
-	if !config.IsCipher {
-		return config.CloudInitUserData, nil
-	}
-	if !config.IsValidCipher {
-		errStr := fmt.Sprintf("%s, Cipher Block is not ready", config.DisplayName)
-		return nil, errors.New(errStr)
+// getCloudInitUserData : returns decrypted cloud-init user data
+func getCloudInitUserData(ctx *domainContext,
+	config types.DomainConfig) (*string, error) {
+	cipherBlock := new(types.CipherBlockStatus)
+	*cipherBlock = config.CipherBlockStatus
+	cipherBlock.CipherBlockID = getCipherBlockID(cipherBlock.Key(),
```

Comment:
What is this ID used for? I can't quite understand why you have the AppNum in there.

Reply:
Removed, the cipherblock will use the config.Key() as ID.